src/ASM/LR/ASMu2Drecovery.C

```diff
@@ -23,8 +23,12 @@
 #include "Utilities.h"
 #include "Profiler.h"
 #include "IntegrandBase.h"
+#include "FiniteElement.h"
 #include <array>
 
+#include <iostream>
```

Comment:
No need to include iostream here, since it is included via other -h files (none of the other C-files in ASM has it, to be consistent. Besides, wrap the inclusion of fstream within #ifdef SP_DEBUG to make it clear it is only needed for debug output. All this pure cosmetics of course.